### PR TITLE
Mobileapps-983

### PR DESCRIPTION
**Pull request by aman-alfresco**
1. add params in advance search api for facet search
2. changes in the bucket and queries models as per response

### CODE REVIEW

File: content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt [hunk 1]
```diff
@@ -1,6 +1,11 @@
 package com.alfresco.content.apis
 
 import com.alfresco.content.models.RequestDefaults
+import com.alfresco.content.models.RequestFacetField
+import com.alfresco.content.models.RequestFacetFields
+import com.alfresco.content.models.RequestFacetIntervals
+import com.alfresco.content.models.RequestFacetIntervalsIntervals
```
Comment (gl-lovekesh) [inline content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt:7]: Correct the name of the class.

File: sample/src/main/java/com/alfresco/sample/MainViewModel.kt [hunk 2]
```diff
@@ -87,10 +93,49 @@ class MainViewModel(private val context: Context) : ViewModel() {
...
+                    RequestFacetSet(label = "lastYear", start = "2018", end = "2019", endInclusive = false),
+                    RequestFacetSet(label = "currentYear", start = "NOW/YEAR", end = "NOW/YEAR+1YEAR"),
+                    RequestFacetSet(label = "earlier", start = "*", end = "2018", endInclusive = false)
+                )
+            ), RequestFacetIntervalsIntervals(
+                label = "TheModified", field = "cm:modified", sets = listOf(
+                    RequestFacetSet(label = "2017", start = "2017", end = "2018", endInclusive = false),
+                    RequestFacetSet(label = "2017-2018", start = "2017", end = "2018", endInclusive = true),
+                    RequestFacetSet(label = "currentYear", start = "NOW/YEAR", end = "NOW/YEAR+1YEAR"),
+                    RequestFacetSet(label = "earlierThan2017", start = "*", end = "2017", endInclusive = false)
```
Comment (gl-lovekesh) [inline sample/src/main/java/com/alfresco/sample/MainViewModel.kt:127]: Localisation is missing

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]
Localization is missing.

File: content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt (hunk 1 above)
Comment (aman-alfresco) [reply content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt]: Resolved

File: sample/src/main/java/com/alfresco/sample/MainViewModel.kt (hunk 2 above)
Comment (aman-alfresco) [reply sample/src/main/java/com/alfresco/sample/MainViewModel.kt:127]: This is for sample app.

File: content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt (hunk 1 above)
Comment (aman-alfresco) [reply content-ktx/src/main/kotlin/com/alfresco/content/apis/SearchApi.kt]: Done

**Pull request merged**